source/mir/random/discrete.d

```diff
@@ -88,36 +78,37 @@ struct Discrete(T)
     size_t opCall(RNG)(ref RNG gen) const
     {
         import std.random : uniform;
-        T v = uniform!("[)", T, T)(0, _cdPoints[$-1], gen);
-        return (cast(SortedRange!(const(T)[], "a <= b")) r).lowerBound(v).length;
+        import std.range : assumeSorted;
+
+        T v = uniform!("[)", T, T)(0, cdPoints[$-1], gen);
```

Comment:
I think elements with zero probs are problematic in both cases - see these two examples:

General example
-----------------------

```d
import std.range;
cdPoints = [1, 3, 4];
foreach (el; 0..5)
    writeln(el, "->", cdPoints.assumeSorted!"a < b".lowerBound(el));
foreach (el; 0..5)
    writeln(el, "->", cdPoints.assumeSorted!"a <= b".lowerBound(el));
```

```
0->[]
1->[]
2->[1]
3->[1]
4->[2] // only included in "[]"

0->[]
1->[1]
2->[1]
3->[1, 3]
4->[1, 3] // only included in "[]"
```

With zero probs
---------------------

```d
import std.range;
cdPoints = [0, 1, 3, 4];
foreach (el; 0..5)
    writeln(el, "->", cdPoints.assumeSorted!"a < b".lowerBound(el));
foreach (el; 0..5)
    writeln(el, "->", cdPoints.assumeSorted!"a <= b".lowerBound(el));
```

```
0->[]
1->[0]
2->[0, 1]
3->[0, 1]
4->[0, 1, 3] // only included in "[]"

0->[0] 
1->[0, 1]
2->[0, 1]
3->[0, 1, 3]
4->[0, 1, 3, 4]  // only included in "[]"
```